api/modelmanager/modelinfo_test.go

```diff
@@ -31,25 +32,42 @@ func (s *modelInfoSuite) assertResponse(c *gc.C, result interface{}) *params.Mod
 	return result.(*params.ModelInfoResults)
 }
 
-func (s *modelInfoSuite) TestModelInfo(c *gc.C) {
-	results := params.ModelInfoResults{
-		Results: []params.ModelInfoResult{{
-			Result: &params.ModelInfo{Name: "name", UUID: "etc."},
-		}, {
-			Error: &params.Error{Message: "woop"},
-		}},
-	}
+func (s *modelInfoSuite) assertExpectedModelInfo(c *gc.C, expectedInfo params.ModelInfoResults) {
 	apiCaller := basetesting.APICallerFunc(
 		func(objType string, version int, id, request string, a, result interface{}) error {
 			s.checkCall(c, objType, id, request)
 			resp := s.assertResponse(c, result)
-			*resp = results
+			*resp = expectedInfo
```

Comment:
The correct answer to your question :D is that..
The test injects mock of apiserver and verifies that api returns result that contains agent version if apiserver passed it along...

It could be considered redundant if we cared about lines of code or note cared about the coverage. The reality is that sometimes we transform apiserver results in api calls. This test is to ensure that agent version does not get lost along the way.